File: src/javaapp/Customer.java
package javaapp;

public class Customer {
    private ShoppingCart shoppingCart;
    private ProductBrowser productBrowser;

    public Customer(ShoppingCart shoppingCart, ProductBrowser productBrowser) {
        this.shoppingCart = shoppingCart;
        this.productBrowser = productBrowser;
    }

    public void viewProductListing() {
        productBrowser.displayProducts();
    }

    public void viewProductDetails(String productId) {
        productBrowser.displayProductDetails(productId);
    }

    public void addToCart(String productId, int quantity) {
        Product product = productBrowser.getProductCatalog().getProductById(productId);
        if(product != null) {
            shoppingCart.addItem(product, quantity);
            System.out.println(quantity + " " + product.getName() + " added to cart");
        } else {
            System.out.println("Product not found");
        }
    }

    public void removeFromCart(String productId) {
        Product product = productBrowser.getProductCatalog().getProductById(productId);
        if(product != null) {
            shoppingCart.removeItem(product);
            System.out.println(product.getName() + " removed from cart");
        } else {
            System.out.println("Product not found");
        }
    }

    public void clearCart() {
        shoppingCart.clearCart();
    }

    public void checkout() {
        shoppingCart.displayCart();
    }
}

File: src/kotlinapp/Customer.kt
package kotlinapp

import javaapp.ProductBrowser
import javaapp.ShoppingCart

class Customer(private val shoppingCart: ShoppingCart, private val productBrowser: ProductBrowser) {
    fun viewProductListing() {
        productBrowser.displayProducts()
    }

    fun viewProductDetails(productId: String?) {
        productBrowser.displayProductDetails(productId)
    }

    fun addToCart(productId: String?, quantity: Int) {
        val product = productBrowser.productCatalog.getProductById(productId)
        if (product != null) {
            shoppingCart.addItem(product, quantity)
            println(quantity.toString() + " " + product.name + " added to cart")
        } else {
            println("Product not found")
        }
    }

    fun removeFromCart(productId: String?) {
        val product = productBrowser.productCatalog.getProductById(productId)
        if (product != null) {
            shoppingCart.removeItem(product)
            println(product.name + " removed from cart")
        } else {
            println("Product not found")
        }
    }

    fun clearCart() {
        shoppingCart.clearCart()
    }

    fun checkout() {
        shoppingCart.displayCart()
    }
}


